src/sentry/integrations/pipeline.py

```diff
@@ -177,3 +178,19 @@ def _dialog_success(self, org_integration):
     def _dialog_response(self, data, success):
         context = {"payload": {"success": success, "data": data}}
         return render_to_response("sentry/integrations/dialog-complete.html", context, self.request)
+
+    def fetch_state(self, key: str | None = None) -> Any | None:
+        step_index = self.state.step_index
+        if step_index >= len(self.pipeline_views):
+            return super().fetch_state(key)
+        view = self.pipeline_views[step_index]
+        if isinstance(view, NestedPipelineView):
+            # Attempt to surface state from a nested pipeline
+            nested_pipeline = view.pipeline_cls(
+                organization=self.organization,
+                request=self.request,
+                provider_key=view.provider_key,
+                config=view.config,
+            )
+            return nested_pipeline.fetch_state(key)
```

Comment:
Why didn’t you just put this in the base pipeline fetch_state?

Reply:
@evanpurkhiser https://github.com/getsentry/sentry/pull/39689